cnf-tests/testsuites/e2esuite/knmstate/knmstate_sriov.go

```diff
@@ -0,0 +1,275 @@
+				resNum := testedNode.Status.Allocatable[corev1.ResourceName("openshift.io/testnmstateresource")]
+				capacity, _ := resNum.AsInt64()
+				return capacity
+			}).
+				WithPolling(5*time.Second).
+				WithTimeout(10*time.Minute).
+				Should(Equal(int64(10)), "the number of the observed VFs is not that same as the one sriov created")
+
+			By("Verify Physical Function IP address has been correctly configured")
+			Eventually(func(g Gomega) {
+				out, err := ipAddrShow(node, testDevice.Name)
+				g.Expect(err).ToNot(HaveOccurred())
+				g.Expect(out).To(ContainSubstring("192.0.2.2"))
+				g.Expect(out).To(ContainSubstring("UP"))
+			}).
+				WithPolling(5 * time.Second).
+				WithTimeout(1 * time.Minute).
+				Should(Succeed())
+		})
+	})
+})
+
+func isKnmstateOperatorInstalled() bool {
+	list := nmstatev1.NMStateList{}
+	err := client.Client.List(context.Background(), &list)
+	if k8serrors.IsNotFound(err) {
+		return false
+	}
+
+	Expect(err).ToNot(HaveOccurred())
+	return true
+}
+
+func ipAddrShow(node, intf string) (string, error) {
+	nodeObj, err := client.Client.Nodes().Get(context.Background(), node, metav1.GetOptions{})
+	if err != nil {
+		return "", fmt.Errorf("can't get node[%s]: %w", node, err)
+	}
+
+	cmd := []string{"chroot", "/rootfs", "ip", "-brief", "address", "show", "dev", intf}
+	buf, err := nodes.ExecCommandOnMachineConfigDaemon(client.Client, nodeObj, cmd)
+	if err != nil {
+		return "", fmt.Errorf("command[%v] failed with output[%s] on node[%s]: %w", cmd, string(buf), node, err)
+	}
+	return string(buf), err
+}
+
+// findNonPrimarySriovDevicesAndNode returns a node name with its SR-IOV devices. The device list does not include
+// the one used for primary networking, if it is an SR-IOV device.
+func findNonPrimarySriovDevicesAndNode(n *cluster.EnabledNodes) (string, []*sriovv1.InterfaceExt, error) {
+	var returnError error
+
+	for _, node := range n.Nodes {
+		devices, err := n.FindSriovDevices(node)
+		if err != nil {
+			returnError = errors.Join(returnError, err)
+			continue
+		}
+
+		mainDeviceName, err := getPrimaryNICForNode(node)
+		if err != nil {
+			returnError = errors.Join(returnError, err)
+			continue
+		}
+
+		nonPrimaryDevices := []*sriovv1.InterfaceExt{}
+		for _, device := range devices {
+			if device.Name == mainDeviceName {
+				continue
+			}
+			nonPrimaryDevices = append(nonPrimaryDevices, device)
+		}
+
+		if len(nonPrimaryDevices) > 0 {
+			return node, nonPrimaryDevices, nil
+		}
+	}
+
+	return "", nil, fmt.Errorf("can't find any SR-IOV devices in cluster's nodes: %w", returnError)
+}
+
+func getPrimaryNICForNode(node string) (string, error) {
+	nodeObj, err := client.Client.Nodes().Get(context.Background(), node, metav1.GetOptions{})
+	if err != nil {
+		return "", err
+	}
+
+	buf, err := nodes.ExecCommandOnMachineConfigDaemon(client.Client, nodeObj,
+		[]string{"/bin/bash", "-c", "chroot /rootfs /usr/bin/ovs-vsctl list-ports br-ex | /usr/bin/grep -v patch"})
```

Comment:
here chroot is needed because ovs-vsctl command does not exist outside the rootfs